### DISCUSSION TITLE
auto-instrumentation disabled for specific framework

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#6323] [Q&A]
Hi, 
Our services are using wide range of tech stack and we have started to use 'auto-instrumentation' option for all these services.
As a result, we are observing too many spans being created (those are not required for us) and a drastic spike in disk space consumption within few hours of auto-instrumentation.

I am aware of the option mentioned [here ](https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#suppressing-specific-agent-instrumentation) to disable the auto-instrumentation for a specific framework / library.
But, do you have any recommendations / analysis that suggest the set of frameworks / libraries those might be too noisy (generates regular spans + low level spans).?
We would like to proactively disable those framework's / library's auto-instrumentation ?

**Comment by mateuszrzeszutek** [comment]
Hey @itsmaneesh ,

> do you have any recommendations / analysis that suggest the set of frameworks / libraries those might be too noisy (generates regular spans + low level spans).?

It really depends on your application. I recommend running your application for a while with the default config, and seeing for yourself which spans are not necessary.
